### PR TITLE
feat: added reconstructing split token logic

### PR DESCRIPTION
**Pull request by gireesh-naidu**

### CODE REVIEW

File: oidc/http.go [hunk 1]
```diff
@@ -197,3 +197,34 @@ func (rt *TransportWithHeader) RoundTrip(r *http.Request) (*http.Response, error
...
+		if len(tokenChunks) == 1 {
+			return token
+		}
+
+		noChunks, err := strconv.Atoi(tokenChunks[0])
+		if err != nil {
+			log.Infof("err in re-constructing splitted token: %v\n", err)
+		}
+
+		token = tokenChunks[1]
```
Comment (nishant-d) [inline oidc/http.go]: length of tokenChunks can be zero?

File: oidc/http.go [hunk 2]
```diff
@@ -197,3 +197,34 @@ func (rt *TransportWithHeader) RoundTrip(r *http.Request) (*http.Response, error
...
+		for i := 1; i < noChunks; i++ {
+			cookieName := AuthCookieName + fmt.Sprintf("-%d", i)
```
Comment (nishant-d) [inline oidc/http.go]: we are using string concatenation and sprint both ? sprint should be enough

**Review by nishant-d — CHANGES_REQUESTED** [review]
add unit test to `ReConstructSplitToken` method.

File: oidc/http.go (hunk 1 above)
Comment (gireesh-naidu) [reply oidc/http.go]: no

**Review by nishant-d — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**